packages/cli/api-importers/v2-importer-commons/src/converters/abstract/AbstractParameterConverter.ts

```diff
@@ -0,0 +1,183 @@
+                        allowMultiple: this.parameter.explode ?? false,
+                        v2Examples: {
+                            userSpecifiedExamples: await this.convertParameterExamples({ context, errorCollector }),
+                            autogeneratedExamples: {}
+                        },
+                        availability
+                    },
+                    inlinedTypes
+                };
+            case "header":
+                return {
+                    type: "header",
+                    parameter: {
+                        name: context.casingsGenerator.generateNameAndWireValue({
+                            name: this.parameter.name,
+                            wireValue: this.parameter.name
+                        }),
+                        docs: this.parameter.description,
+                        valueType: typeReference ?? AbstractParameterConverter.OPTIONAL_STRING,
+                        env: undefined,
+                        v2Examples: {
+                            userSpecifiedExamples: await this.convertParameterExamples({ context, errorCollector }),
+                            autogeneratedExamples: {}
+                        },
+                        availability
+                    },
+                    inlinedTypes
+                };
+            case "path":
+                return {
+                    type: "path",
+                    parameter: {
+                        name: context.casingsGenerator.generateName(this.parameter.name),
+                        docs: this.parameter.description,
+                        valueType: typeReference ?? AbstractParameterConverter.STRING,
+                        location: "ENDPOINT",
+                        variable: undefined,
+                        v2Examples: {
+                            userSpecifiedExamples: await this.convertParameterExamples({ context, errorCollector }),
+                            autogeneratedExamples: {}
+                        }
+                    },
+                    inlinedTypes
+                };
+            default:
+                return undefined;
+        }
+    }
+
+    protected async convertParameterExamples({
+        context,
+        errorCollector
+    }: {
+        context: AbstractConverterContext<object>;
+        errorCollector: ErrorCollector;
+    }): Promise<Record<string, unknown>> {
+        let userSpecifiedExamples: Record<string, unknown> = {};
+        const parameterExample = this.parameter.example;
+        const parameterExamples = this.parameter.examples;
+        for (const [key, example] of Object.entries(parameterExamples ?? {})) {
+            if (context.isReferenceObject(example)) {
+                const resolved = await context.resolveReference(example);
+                if (resolved.resolved) {
+                    if (typeof resolved.value === "object" && resolved.value != null && "value" in resolved.value) {
```

Comment:
nit: typically its nice to put these sort of checks in a helper function and name it `isExampleWithValueAndName` or something